### PR TITLE
Rustfmt layout crate

### PR DESCRIPTION
**Pull request by pyfisch**
Part of #21373

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21509)

### CODE REVIEW

File: components/layout/query.rs [hunk 1]
```diff
@@ -780,15 +843,20 @@ where
             layout_el: <N::ConcreteThreadSafeLayoutNode as ThreadSafeLayoutNode>::ConcreteThreadSafeLayoutElement,
             layout_root: &mut Flow,
             requested_node: N,
-            longhand_id: LonghandId) -> String {
```
Comment (pyfisch) [inline components/layout/query.rs]: Rustfmt really struggled here. It wants to format the line as `longhand_id: LonghandId) -> String{` without any indentation. I manually formatted it so both rustfmt and tidy accept it.

Comment (jdm) [reply components/layout/query.rs]: Could you recreate a minimal testcase and submit an issue to https://github.com/rust-lang-nursery/rustfmt/?

**Comment by jdm** [comment]
Note to self: continue reviewing starting from components/layout/table.rs.

**Review by jdm — APPROVED** [review]

**Comment by jdm** [comment]
This is good to merge once it's rebased.
@bors-servo delegate+

**Comment by pyfisch** [comment]
Rebased. I apparently have updated rustfmt in the meantime. Therefore the formatting is slightly different in a few places. I checked all changes.

There a two kinds of changes. A short method call is placed on the same line as }). 

```diff
-            })
-            .collect();
+            }).collect();
```

The other change is done in only two or three places and splits a comparison across multiple lines which I think looks odd.

```diff
-                .visibility != Visibility::Visible
+                .visibility !=
+                Visibility::Visible
```

@bors-servo r=jdm

**Comment by jdm** [comment]
Thanks for the summary!

**Pull request merged**